hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -366,33 +367,29 @@ public void deleteContainer(long containerID) throws IOException {
   }
 
   @Override
-  public void notifyObjectStageChange(StorageContainerLocationProtocolProtos
-      .ObjectStageChangeRequestProto.Type type, long id,
-      StorageContainerLocationProtocolProtos.ObjectStageChangeRequestProto.Op
-          op, StorageContainerLocationProtocolProtos
-      .ObjectStageChangeRequestProto.Stage stage) throws IOException {
-
-    LOG.info("Object type {} id {} op {} new stage {}", type, id, op,
-        stage);
-    if (type == StorageContainerLocationProtocolProtos
-        .ObjectStageChangeRequestProto.Type.container) {
-      if (op == StorageContainerLocationProtocolProtos
-          .ObjectStageChangeRequestProto.Op.close) {
-        if (stage == StorageContainerLocationProtocolProtos
-            .ObjectStageChangeRequestProto.Stage.begin) {
-          scm.getContainerManager()
-              .updateContainerState(ContainerID.valueof(id),
-                  HddsProtos.LifeCycleEvent.FINALIZE);
-        } else {
-          scm.getContainerManager()
-              .updateContainerState(ContainerID.valueof(id),
-                  HddsProtos.LifeCycleEvent.CLOSE);
-        }
+  public void closeContainer(long containerID) throws IOException {
+    final String remoteUser = getRpcRemoteUsername();
+    final Map<String, String> auditMap = Maps.newHashMap();
+    auditMap.put("containerID", String.valueOf(containerID));
+    auditMap.put("remoteUser", remoteUser);
+    try {
+      scm.checkAdminAccess(remoteUser);
+      final ContainerID cid = ContainerID.valueof(containerID);
+      final HddsProtos.LifeCycleState state = scm.getContainerManager()
+          .getContainer(cid).getState();
+      if (!state.equals(HddsProtos.LifeCycleState.OPEN)) {
+        throw new SCMException("Cannot close an " + state + " container.",
```

Comment:
```suggestion
        throw new SCMException("Cannot close a " + state + " container.",
```

Currently all non-`OPEN` state names start with a consonant, so "a" is more appropriate than "an", eg. "Cannot close a CLOSED container".  I only mention this because the message appears in the command output.

Reply:
Thanks for the catch.